File: week2/problem set/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include<string.h>
#include<math.h>
#include<ctype.h>

int letterCount(string text);
int wordCount(string text);
int sentCount(string text);

int main(void)
{
    string text = get_string("Text: ");
    int letters_sum = letterCount(text);
    int words_sum = wordCount(text);
    int sentence_sum = sentCount(text);

    float S = (float)sentence_sum / words_sum * 100;
    float L = (float)letters_sum / words_sum * 100;

    int index = round(0.0588 * L - 0.296 * S -15.8);

    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n",index);
    }
}

int letterCount(string text)
{
    int count = 0;
    int i = 0;
    do
    {
        if(isalpha(text[i]))
        {
            count++;
        }
        i++;
    }
    while(i < strlen(text));

    return count;
}

int wordCount(string text)
{
    int count = 1;
    int i = 0;
    /* for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            count++;
        }
    } */
    do
    {
        if (text[i] == ' ')
        {
            count++;
        }
        i++;
    }
    while (i < strlen(text));
    return count;
}

int sentCount(string text)
{
    int count = 0;
    int i = 0;

    while (text[i] != '\0')
    {
        if(text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count++;
        }
        i++;
    }
    return count;
}

File: week6/problem set/sentimental-readability/readability.py
# TODO
from cs50 import get_string

def main():
    text = get_string("Text: ")
    letters, words, sentences = count_elements(text)

    L = (letters / words) * 100
    S = (sentences / words) * 100
    index = 0.0588 * L - 0.296 * S - 15.8

    grade_level = round(index)

    if grade_level >= 16:
        print("Grade 16+")
    elif grade_level < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {grade_level}")

def count_elements(text):
    letters = 0
    words = 1  # Initialize to 1 to account for the last word without a space after it
    sentences = 0

    for char in text:
        if char.isalpha():
            letters += 1
        elif char.isspace():
            words += 1
        elif char in ['.', '!', '?']:
            sentences += 1

    return letters, words, sentences

if __name__ == "__main__":
    main()